helix-dap/src/transport.rs

```diff
@@ -0,0 +1,265 @@
+use crate::{Error, Result};
+use anyhow::Context;
+use log::{error, info, warn};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::{
+    io::{AsyncBufRead, AsyncBufReadExt, AsyncReadExt, AsyncWrite, AsyncWriteExt},
+    sync::{
+        mpsc::{unbounded_channel, Sender, UnboundedReceiver, UnboundedSender},
+        Mutex,
+    },
+};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Request {
+    #[serde(skip)]
+    pub back_ch: Option<Sender<Result<Response>>>,
+    pub seq: u64,
+    #[serde(rename = "type")]
+    pub msg_type: String,
+    pub command: String,
+    pub arguments: Option<Value>,
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+pub struct Response {
+    pub seq: u64,
```

Comment:
Delve sets 0 here (no counter), but `codelldb` omitted this field at all, although `seq` is mandatory. Since I see no sense in tracking those, I'll remove seq from response and event structs rather than specifying a quirk.